### PR TITLE
Setup Karma + Jasmine, run tests on Chrome and Firefox

### PR DESCRIPTION
**Pull request by limonte**
Connected to #18 

Initial setup of Karma (test runner) + Jasmine (testing framework).

Since we're using ES6 modules (`import`/`export`), we need to compile stuff. ~For that `webpack` + `babel-loader` are used.~ UPD: `rollup-preprocessor` is used now.

Karma has shown itself extremely well on my most popular open-sourced project - [SweetAlert2](https://github.com/sweetalert2/sweetalert2), we are running automated tests against all supported browsers almost for free because there are karma plugins for IE, SauceLabs, Browserstack, etc.

 - IE11 with AppVeyour: https://ci.appveyor.com/project/limonte/sweetalert2/build/495
 - Android, iPhone, Safari, Edge with SauceLabs: https://travis-ci.org/sweetalert2/sweetalert2/builds/380682532 

---

Before merging this PR, you need to go to Travis and turn on CI testing for this project.

This is how CI build for `css-doodle` will look like https://travis-ci.org/limonte/css-doodle/builds/380677371

Have a great day and test your awesome stuff! :)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,15 +6,26 @@
...
+    "test": "karma start --single-run"
   },
   "repository": {
     "type": "git",
     "url": "git+https://github.com/yuanchuan/css-doodle.git"
   },
   "devDependencies": {
+    "babel-core": "^6.26.3",
+    "babel-loader": "^7.1.4",
+    "gulp": "^3.9.1",
```
Comment (yuanchuan) [inline package.json]: It seems the gulp package hasn't been used?

Comment (limonte) [reply package.json]: Oops, forgot to remove it after some experimenting.

Removed.

**Comment by yuanchuan** [comment]
All looking good except the overlap functionality of `babel + webpack`  and `rollup` (which is currently used for the ES6 thing).

So I tried on my local machine with these modifications to the `package.json`:

```
-    "babel-core": "^6.26.3",
-    "babel-loader": "^7.1.4",
     "jasmine-core": "^3.1.0",
     "karma": "^2.0.2",
     "karma-chrome-launcher": "^2.2.0",
     "karma-firefox-launcher": "^1.1.0",
     "karma-jasmine": "^1.1.2",
-    "karma-webpack": "^3.0.0",
+    "karma-rollup-preprocessor": "^6.0.0",
     "rollup": "^0.58.1",
-    "uglify-es": "^3.3.9",
-    "webpack": "^4.8.3"
+    "uglify-es": "^3.3.9"
```

And with new `karma.conf.js` base on yours:

```
module.exports = function(config) {
  config.set({
    browsers: ['ChromeHeadless', 'Firefox'],

    frameworks: ['jasmine'],

    files: [
      'test/*.js'
    ],

    preprocessors: {
      'src/**/*.js': ['rollup'],
      'test/**/*.js': ['rollup'],
    },

    rollupPreprocessor: {
      output: {
        format: 'iife',
        sourcemap: 'inline'
      }
    }

  })
}
```

It is more lightweight now!  How do you think about this?

**Comment by limonte** [comment]
Absolutely yes! Less dependencies and less code => less bugs :)

Changed. Oh, you enabled Travis already :tada:

**Pull request merged**